src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/PublicSymbols/AnonymousType.TypePublicSymbol.cs

```diff
@@ -22,7 +22,7 @@ internal sealed partial class AnonymousTypeManager
         /// Represents an anonymous type 'public' symbol which is used in binding and lowering.
         /// In emit phase it is being substituted with implementation symbol.
         /// </summary>
-        private sealed class AnonymousTypePublicSymbol : NamedTypeSymbol
+        internal sealed class AnonymousTypePublicSymbol : NamedTypeSymbol
```

Comment:
Perhaps not, but the code would be more complex. Also the corresponding type is already internal in VB.